File: 卡码/代码随想录/1/09-动态规划/Js/04-股票问题/LeetCode_0121.js
//121. 买卖股票的最佳时机
const maxProfit = function(prices) {
	const dp=[prices[0],0];
	for(let i=1;i<=prices.length;i++){
		dp[0]=Math.min(dp[0],prices[i-1]);
		dp[1]=Math.max(dp[1],prices[i-1]-dp[0]);
	}
	return dp[1];
};

File: 卡码/代码随想录/1/09-动态规划/Go/04-股票问题/LeetCode_0121.go
// 121. 买卖股票的最佳时机
package main

import (
	"math"
)

func maxProfit(prices []int) int {
	dp := []int{prices[0], 0}
	for i := 1; i <= len(prices); i++ {
		dp[0] = int(math.Min(float64(dp[0]), float64(prices[i-1])))
		dp[1] = int(math.Max(float64(dp[1]), float64(prices[i-1]-dp[0])))
	}
	return dp[1]
}
